src/operation-node/alter-column-node.ts

```diff
@@ -26,20 +24,11 @@ export const AlterColumnNode = freeze({
     return node.kind === 'AlterColumnNode'
   },
 
-  create(column: string): AlterColumnNode {
+  create<T extends keyof AlterColumnNode>(column: string, prop: T, value: Required<AlterColumnNode>[T]): AlterColumnNode {
```

Comment:
You're allowing setting `'kind'` and `'column'` as prop.

```suggestion
  create<P extends keyof AlterColumnNodeProps>(column: string, prop: P, value: Required<AlterColumnNode>[P]): AlterColumnNode {
```